src/networking.c

```diff
@@ -539,6 +539,116 @@ get_mac_address(const gchar *const ifname)
 	return macaddr;
 }
 
+/*
+ * Get and store the iptable rules set in the
+ * network namespace. These need to be re-applied
+ * within the container.
+ *
+ * \param config \ref cc_oci_config
+ *
+ * \return \c true on success, else \c false.
+ */
+static gboolean
+get_iptable_rules(struct cc_oci_config *config)
+{
+	gboolean ret;
+	GError *error = NULL;
+	gint exit_status = 0;
+	char *cmd = "iptables-save";
+	char *args[] = { cmd, NULL };
+
+	if (! config) {
+		return false;
+	}
+
+	g_debug("Running command : %s", cmd);
+	ret = g_spawn_sync (NULL,
+              args,
+              NULL,
+              G_SPAWN_SEARCH_PATH | G_SPAWN_STDERR_TO_DEV_NULL,
+              NULL,
+              NULL,
+              &(config->net.iptable_rules),
+              NULL,
+              &exit_status,
+              &error);
+
+	if ( !ret || exit_status) {
+		if (! ret) {
+			g_critical("Error spawning process for %s:%s",
+				cmd, error->message);
+			g_clear_error(&error);
+		}
+
+		if (! g_spawn_check_exit_status(exit_status, &error)) {
+			g_critical("Error running command %s: %s",
+				cmd, error->message);
+			g_clear_error(&error);
+		}
+		g_free_if_set(config->net.iptable_rules);
+	}
+
+	return ret;
+}
+
+/*
+ * Flush all iptable rules set in network namespace.
+ * We need to do this as rules get applied to the
+ * packets crossing the tap bridge.
+ *
+ * \return \c true on success, else \c false.
+ */
+static gboolean
+purge_iptable_rules(void)
+{
+	gboolean ret;
+	GError *error = NULL;
+	gint exit_status = 0;
+
+	/* Set the default policies for the built-in chains to ACCEPT,
+	 * flush nat and mangle tables, flush all chains(-F) and
+	 * delete all user chains(-X).
+	 */
+	char *iptables_drop_command = "iptables -P INPUT ACCEPT "
```

Comment:
@jodh-intel I avoided this as g_spawn_async expects non-const arguments, passing const gives a bunch of warnings